packages/flutter/lib/src/cupertino/action_sheet.dart

```diff
@@ -0,0 +1,1264 @@
+  _CupertinoAlertRenderWidget get widget => super.widget;
+
+  @override
+  _RenderCupertinoAlert get renderObject => super.renderObject;
+
+  @override
+  void visitChildren(ElementVisitor visitor) {
+    if (_contentElement != null) {
+      visitor(_contentElement);
+    }
+    if (_actionsElement != null) {
+      visitor(_actionsElement);
+    }
+  }
+
+  @override
+  void mount(Element parent, dynamic newSlot) {
+    super.mount(parent, newSlot);
+    _contentElement = updateChild(_contentElement,
+        widget.contentSection, _AlertSections.contentSection);
+    _actionsElement = updateChild(_actionsElement,
+        widget.actionsSection, _AlertSections.actionsSection);
+  }
+
+  @override
+  void insertChildRenderObject(RenderObject child, _AlertSections slot) {
+    _placeChildInSlot(child, slot);
+  }
+
+  @override
+  void moveChildRenderObject(RenderObject child, _AlertSections slot) {
+    _placeChildInSlot(child, slot);
+  }
+
+  @override
+  void update(RenderObjectWidget newWidget) {
+    super.update(newWidget);
+    _contentElement = updateChild(_contentElement,
+        widget.contentSection, _AlertSections.contentSection);
+    _actionsElement = updateChild(_actionsElement,
+        widget.actionsSection, _AlertSections.actionsSection);
+  }
+
+  @override
+  void forgetChild(Element child) {
+    assert(child == _contentElement || child == _actionsElement);
+    if (_contentElement == child) {
+      _contentElement = null;
+    } else if (_actionsElement == child) {
+      _actionsElement = null;
+    }
+  }
+
+  @override
+  void removeChildRenderObject(RenderObject child) {
+    assert(child == renderObject.contentSection || child == renderObject.actionsSection);
+    if (renderObject.contentSection == child) {
+      renderObject.contentSection = null;
+    } else if (renderObject.actionsSection == child) {
+      renderObject.actionsSection = null;
+    }
+  }
+
+  void _placeChildInSlot(RenderObject child, _AlertSections slot) {
+    assert(slot != null);
+    switch (slot) {
+      case _AlertSections.contentSection:
+        renderObject.contentSection = child;
+        break;
+      case _AlertSections.actionsSection:
+        renderObject.actionsSection = child;
+        break;
+    }
+  }
+}
+
+// iOS style layout policy for sizing an alert's content section and action
+// button section.
+//
+// The policy is as follows:
+//
+// If all content and buttons fit on screen:
+// The content section and action button section are sized intrinsically.
+//
+// If all content and buttons do not fit on screen:
+// A minimum height for the action button section is calculated. The action
+// button section will not be rendered shorter than this minimum.  See
+// [_RenderCupertinoAlertActions] for the minimum height calculation.
+//
+// With the minimum action button section calculated, the content section is
+// laid out as tall as it wants to be, up to the point that it hits the
+// minimum button height at the bottom.
```

Comment:
"up to the point that it hits the minimum button height at the bottom" -- I'm not sure how to parse that.  Can you reword it?